rotkehlchen/chain/gnosis/modules/sdai/decoder.py

```diff
@@ -0,0 +1,109 @@
+    def __init__(
+            self,
+            gnosis_inquirer: 'GnosisInquirer',
+            base_tools: 'BaseDecoderTools',
+            msg_aggregator: 'MessagesAggregator',
+    ) -> None:
+        super().__init__(
+            evm_inquirer=gnosis_inquirer,
+            base_tools=base_tools,
+            msg_aggregator=msg_aggregator,
+        )
+        self.sdai = A_SDAI.resolve_to_evm_token()
+
+    def _decode_sdai_deposit_events(self, context: DecoderContext) -> DecodingOutput:
+        amount_raw = hex_or_bytes_to_int(context.tx_log.data[:32])
+        amount = token_normalized_value_decimals(amount_raw, None)
+        shares_raw = hex_or_bytes_to_int(context.tx_log.data[32:])
+        shares = token_normalized_value_decimals(shares_raw, None)
+
+        for event in context.decoded_events:
+            if (
+                event.event_type == HistoryEventType.SPEND and
+                event.event_subtype == HistoryEventSubType.NONE and
+                event.balance.amount == amount
+            ):
+                asset = event.asset.resolve_to_crypto_asset()
+                event.counterparty = CPT_SDAI
+                event.event_type = HistoryEventType.DEPOSIT
+                event.event_subtype = HistoryEventSubType.DEPOSIT_ASSET
+                event.notes = f'Deposit {event.balance.amount} {asset.name} to SDAI contract'
+            elif (
+                event.event_type == HistoryEventType.RECEIVE and
+                event.event_subtype == HistoryEventSubType.NONE and
+                event.balance.amount == shares
+            ):
+                event.event_subtype = HistoryEventSubType.RECEIVE_WRAPPED
+                event.notes = f'Receive {event.balance.amount} SDAI'
+                event.counterparty = CPT_SDAI
+
+        return DEFAULT_DECODING_OUTPUT
+
+    def _decode_sdai_redeem_events(self, context: DecoderContext) -> DecodingOutput:
+        shares_raw = hex_or_bytes_to_int(context.tx_log.data[32:])
+        shares = token_normalized_value_decimals(shares_raw, None)
+
+        for event in context.decoded_events:
+            if (
+                event.event_type == HistoryEventType.SPEND and
+                event.event_subtype == HistoryEventSubType.NONE and
+                event.balance.amount == shares
+            ):
+                event.event_type = HistoryEventType.WITHDRAWAL
+                event.event_subtype = HistoryEventSubType.REMOVE_ASSET
+                event.notes = f'Redeem {event.balance.amount} SDAI'
+                event.counterparty = CPT_SDAI
+
+        return DEFAULT_DECODING_OUTPUT
+
+    def decode_sdai_events(self, context: DecoderContext) -> DecodingOutput:
+        if context.tx_log.topics[0] == SDAI_DEPOSIT:
+            return self._decode_sdai_deposit_events(context)
+        elif context.tx_log.topics[0] == SDAI_REDEEM:
+            return self._decode_sdai_redeem_events(context)
+        else:
+            return DEFAULT_DECODING_OUTPUT
```

Comment:
I believe the linter tells you but you can do

```suggestion
            return self._decode_sdai_redeem_events(context)
        return DEFAULT_DECODING_OUTPUT
```